src/main/java/de/tum/in/www1/artemis/service/iris/IrisSessionService.java

```diff
@@ -55,6 +55,18 @@ public void checkHasAccessToIrisSession(IrisSession session, User user) {
         }
     }
 
+    /**
+     * Checks if the programming exercise for which an Iris operation was requested has Iris activated.
+     * An Iris operation can be performed if the programming exercise has Iris activated.
+     *
+     * @param programmingExercise The programming exercise to check
+     */
+    public void checkIrisActivated(ProgrammingExercise programmingExercise) {
```

Comment:
I think naming it `checkIrisIsActivated` or `checkIsIrisActivated` might make the method nicer to read when used alongside methods like `checkHasAccessToIrisSession`.
But this isn't a must have. You can also leave it as is.